tokens/tsm-report.go

```diff
@@ -0,0 +1,37 @@
+package tokens
+
+import (
+	"encoding/json"
+
+	"github.com/fxamacker/cbor/v2"
+)
+
+const TSMReportMediaType = "application/vnd.veraison.tsm-report+cbor"
+
+type TSMReport struct {
+	AuxBlob       []byte           `json:"auxblob,omitempty"`
+	OutBlob       []byte           `json:"outblob"`
+	Provider      string           `json:"provider"`
+	ServiceReport TsmServiceReport `json:"service,omitempty"`
+}
+
+type TsmServiceReport struct {
+	ManifestBlob    []byte `json:"manifestblob,omitempty"`
+	ServiceProvider string `json:"service_provider,omitempty"`
+}
+
+func (t *TSMReport) ToJSON() ([]byte, error) {
+	return json.Marshal(t)
+}
+
+func (t *TSMReport) FromJSON(data []byte) error {
+	return json.Unmarshal(data, t)
+}
```

Comment:
We only produce the `tsm-report` data items; we don’t need to decode them.

```suggestion
```

Reply:
The server would need this method to de-serialize the evidence, so I prefer to keep it.

Reply:
OK, then you need to add a custom unmarshaler too for the `BinaryString` type, e.g.:

```go
func (o *BinaryString) UnmarshalJSON(b []byte) error {
	var s string

	if err := json.Unmarshal(b, &s); err != nil {
		return err
	}

	tmp, err := base64.RawURLEncoding.DecodeString(s)
	if err != nil {
		*o = BinaryString(tmp)
	}

	return err
}
```

Reply:
Yup, thank you!

Reply:
> 	if err := json.Unmarshal(b, &s); err != nil {
> 		return err
> 	}

Why is this necessary? We could directly cast the input byte slice as string.

Reply:
FYI, there's a typo in the conditional expression below:

> 	if err != nil {
> 		*o = BinaryString(tmp)
> 	}

Reply:
Apologies, I am not sure I understand your question.  That said, there are probably better/more compact ways to do that than the one I put forward.  And as long as it works, I am OK with any solution :-)

Reply:
```golang
func (o *BinaryString) UnmarshalJSON(b []byte) error {
	decoded, err := base64.RawURLEncoding.DecodeString(string(b))
	if err != nil {
		return err
	}

	*o = decoded
	return nil
}
```

I don't know why you were performing the type conversion when `base64.RawURLEncoding.DecodeString` throws an error. I'm going with the above.

Reply:
hmm, that would not work because what `UnmarshalJSON` sees as input, in this case, is a JSON string, i.e., [`'"' characters '"'`](https://www.json.org/json-en.html).

So, you’d need to remove the quoting first, then process it as b64u.  Otherwise, `DecodeString` will choke on the very first char (`"`) which is not in the base64 alphabet.